doc/src/hexdoc/_templates/macros/recipes.html.jinja

```diff
@@ -113,4 +113,105 @@
       {% endfor %}
     </div>
   </details>
-{%- endmacro %}
\ No newline at end of file
+{%- endmacro %}
+
+{% macro render_mindcipe(recipe) -%}
+  <details class="details-collapsible flay-recipe-details">
+    <summary class="collapse-details">
+      <span class="collapse-recipe-show">{{ _('hexdoc.recipe.show') }}</span>
+      <span class="collapse-recipe-hide">{{ _('hexdoc.recipe.hide') }}</span>
+    </summary>
+    {# <div class="flay-recipes">
+      {% for recipe in recipes %} #}
+        <div>
+          <h5>{{ recipe.result.name.name }}</h5>
+          <div class="flay-recipe">
+            <img
+              class="flay-recipe-background"
+              alt="Mindflay recipe gui"
+              src="{{ 'hexcasting:textures/gui/hexdoc/brainsweep.png'|hexdoc_texture }}"
+            >
+            {{ render_brainsweepee(recipe) }}
+            <div class="flay-recipe-block-ingredient">
+              {{ render_item(recipe.blockIn.block) }}
+            </div>
+
+            <div class="flay-recipe-result">
+              {{ render_item(recipe.result.name) }}
+            </div>
+          </div>
+        </div>
+      {# {% endfor %}
+    </div> #}
+  </details>
+{%- endmacro %}
+
+{% macro render_brainsweepee(recipe) -%}
+  {% if recipe.entityIn is defined %}
+    {# 0.11 any entity version #}
+    {% if recipe.entityIn.profession is defined %}
+      {# still a villager #}
+      {{ render_villager(recipe.entityIn) }}
+    {% elif recipe.entityIn.entityType is defined%}
+      {# some actual entity, probably an allay for now atleast #}
+      {{ render_other_entity(recipe.entityIn) }}
+    {% endif %}    
+  {% elif recipe.villagerIn is defined %}
+    {# 0.10 villager version #}
+    {{ render_villager(recipe.villagerIn) }}
+  {% endif %}
+{%- endmacro %}
+
+{% macro render_villager(villager) -%}
+  {% if villager.profession == none %}
+    {% set modid = "minecraft" %}
+    {% set prof = "none" %}
+  {% else %}
+    {% set modid = villager.profession.namespace %}
+    {% set prof = villager.profession.path %}
+  {% endif %} 
```

Comment:
this logic should really be done on the python side imo, so we get full type checking etc. `set` should be an absolute last resort - python should do logic, jinja should stick to filling placeholders wherever possible